mesh/mesh-adapter-api.h

```diff
@@ -0,0 +1,47 @@
+/*
+ *
+ *  BlueZ - Bluetooth protocol stack for Linux
+ *
+ *  Copyright (C) 2020  Silvair Inc. All rights reserved.
+ *
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ */
+#include <ell/dbus.h>
+#include <json-c/json.h>
+
+struct mesh_adpt;
+struct mesh_adpt_private;
+
+
+typedef bool (*mesh_adpt_api_reg_dbus_ifaces)(struct l_dbus *bus);
```

Comment:
These names suck ;)

Reply:
Also, we should be taking adapter as the first argument, following 'object oriented C' conventions.